src/psfmachine/machine.py

```diff
@@ -617,21 +633,45 @@ def _time_bin(self, npoints=200):
             The binned flux, raw
         flux_binned: np.ndarray
             The binned flux, whitened by the mean of the flux in time
-        flux_err_binned:
+        flux_err_binned:: np.ndarray
             The binned flux error, whitened by the mean of the flux
+        pc1_smooth and pc2_smooth: np.ndarray
+            Smooth version of the median poscorr vectors 1 and 2 used for time correction.
+        pc1_bin and pc2_bin: np.ndarray
+            Binned version of poscorrs vectors 1 and 2 used for setting knots.
         """
 
-        # Where there are break points in the data
+        # Where there are break points in the time array
         splits = np.append(
-            np.append(0, np.where(np.diff(self.time) > 0.1)[0]), len(self.time)
+            np.append(0, np.where(np.diff(self.time) > 0.1)[0] + 1), len(self.time)
         )
+        # if using poscorr, find and add discontinuity in poscorr data
+        if hasattr(self, "pos_corr1") and self.time_corrector == "pos_corr":
+            # take the scene-median poscorr
+            mpc1 = np.nanmedian(self.pos_corr1, axis=0)
+            mpc2 = np.nanmedian(self.pos_corr2, axis=0)
+
+            # find poscorr discontinuity in each axis
+            grads1 = np.gradient(mpc1, self.time)
+            grads2 = np.gradient(mpc2, self.time)
+            splits1 = np.where(grads1 > 7 * grads1.std())[0]
+            splits2 = np.where(grads2 > 7 * grads2.std())[0]
+            # merging breaks
+            splits = np.unique(np.concatenate([splits, splits1[1::2], splits2[1::2]]))
+            del grads1, grads2, splits1, splits2
+
         splits_a = splits[:-1] + 100
         splits_b = splits[1:]
         dsplits = (splits_b - splits_a) // npoints
-        breaks = []
+        # breake has the bin centers
+        # first break is at cadance 50
+        breaks = [50]
         for spdx in range(len(splits_a)):
             breaks.append(splits_a[spdx] + np.arange(0, dsplits[spdx] - 1) * npoints)
+        # we include 50-to-last cadecence
+        breaks.append(splits[-1] - 50)
         breaks = np.hstack(breaks)
+        # breaks = np.unique(np.hstack([breaks, splits[1:-1]]))
```

Comment:
I think you can maybe the previous few lines and check there's not a simpler implementation